### PR TITLE
Make benchmark metadata specify benchmark name.

### PR DESCRIPTION
**Pull request by skschneider**
Move benchmark_uid to separate metadata.

### CODE REVIEW

**Comment by skschneider** [comment]
Some unit tests failed. Not sure why tox didn't catch it on my system. Will fix.

**Comment by skschneider** [comment]
Unit tests should be fine now.

File: perfkitbenchmarker/benchmark_spec.py [hunk 1]
```diff
@@ -390,7 +392,8 @@ def PrepareVm(self, vm):
     logging.info('Waiting for boot completion.')
     for port in vm.remote_access_ports:
       vm.AllowPort(port)
-    vm.AddMetadata(benchmark=self.uid, perfkit_uuid=self.uuid)
+    vm.AddMetadata(benchmark=self.name, perfkit_uuid=self.uuid,
+                   benchmark_uid=self.uid)
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmark_spec.py]: I'm a bit ambivalent about including benchmark_uid as metadata - whatever you prefer.

**Comment by cmccoy** [comment]
LGTM.

File: perfkitbenchmarker/benchmark_spec.py (hunk 1 above)
Comment (skschneider) [inline perfkitbenchmarker/benchmark_spec.py]: Now that benchmark name & PKB run_uri don't uniquely identify a benchmark run, it's still useful for identifying resources attached to the same benchmark run. Doesn't really hurt to my knowledge.

**Pull request merged**